### PR TITLE
make PageSizeMatcher and OffsetMatcher optional

### PR DESCRIPTION
**Pull request by Phill101**
pageable routes without any query parameters should return response with default offset and page size

### CODE REVIEW

File: src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala [hunk 1]
```diff
@@ -79,10 +79,10 @@ class UserEndpoints[F[_]: Effect, A] extends Http4sDsl[F] {
 
   private def listEndpoint(userService: UserService[F]): HttpRoutes[F] =
     HttpRoutes.of[F] {
-      case GET -> Root / "users" :? PageSizeMatcher(pageSize) :? OffsetMatcher(offset) =>
+      case GET -> Root / "users" :? OptionalPageSizeMatcher(pageSize) :? OptionalOffsetMatcher(offset) =>
         for {
-          retrived <- userService.list(pageSize, offset)
-          resp <- Ok(retrived.asJson)
+          retrieved <- userService.list(pageSize.getOrElse(10), offset.getOrElse(0))
+          resp <- Ok(retrieved.asJson)
```
Comment (zakpatterson) [inline src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala]: much better spelling.

**Comment by zakpatterson** [comment]
Can we add a new functional test showing the /pets and /users routes work now?

**Comment by Phill101** [comment]
> Can we add a new functional test showing the /pets and /users routes work now?
I'll do it tomorrow

**Comment by pauljamescleary** [comment]
I like it, @zakpatterson

**Comment by zakpatterson** [comment]
I added a commit to this PR at: https://github.com/Phill101/scala-pet-store/pull/1, @Phill101 I think you just have to approve that commit and it will show up here.

File: functional_test/pet_store_client.py [hunk 2]
```diff
@@ -9,6 +9,17 @@
 from urlparse import parse_qs
 from urlparse import urlsplit
 
+def map_to_query_params(m, *args):
```
Comment (pauljamescleary) [inline functional_test/pet_store_client.py]: I have not tested this, but it appears as though this kind of mapping is already present in the python `requests` library and it is possible I have been doing it all wrong?

The following is from the requests documentation.  Seemingly you can do `self.session.request(..., params=your_dictionary_here)`.  Note I didn't try it yet

```
        :param params: (optional) Dictionary or bytes to be sent in the query
            string for the :class:`Request`.
```

Comment (pauljamescleary) [reply functional_test/pet_store_client.py]: Just confirmed this does work.  I think we should use this method, and perhaps a different PR go back and fix all the other methods...

```python
    def list_pets(self, page_size=10, offset=0):
        """
        Returns a list of pets
        :return:
        """
        url = urljoin(self.index_url, "/pets")
        return self.session.request(method='GET', url=url, headers=self.headers, params={'pageSize': page_size, 'offset': offset})
```

Comment (zakpatterson) [reply functional_test/pet_store_client.py]: Hey @Phill101 I added another commit to your PR that does what Paul is talking about here. Just merge that and this should be resolved. https://github.com/Phill101/scala-pet-store/pull/2 

Only problem is the way I've done it the caller of the client method can pass any parameters and they'll be put into the query string. I agree with Paul tho let's revise all these methods for parameter usage in one later PR.

**Pull request merged**